pkg/stanza/operator/helper/parser.go

```diff
@@ -125,6 +125,32 @@ func (p *ParserOperator) ProcessWithCallback(ctx context.Context, entry *entry.E
 	return p.Write(ctx, entry)
 }
 
+func (p *ParserOperator) ParseWithCallback(ctx context.Context, entry *entry.Entry, parse ParseFunction, cb func(*entry.Entry) error) error {
+	// Short circuit if the "if" condition does not match
+	skip, err := p.Skip(ctx, entry)
+	if err != nil {
+		return p.HandleEntryError(ctx, entry, err)
+	}
+	if skip {
+		return p.Write(ctx, entry)
+	}
+
+	if err = p.ParseWith(ctx, entry, parse); err != nil {
+		if p.OnError == DropOnErrorQuiet || p.OnError == SendOnErrorQuiet {
```

Comment:
I'm not sure if we should discard the error here. This could lead to duplicating entries as described at https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35973.

Otherwise, the container parser will get no error back and will further move the entry to the recombine parser.